readers/mutation_readers.cc

```diff
@@ -411,15 +411,19 @@ flat_mutation_reader_v2 make_nonforwardable(flat_mutation_reader_v2 r, bool sing
         flat_mutation_reader_v2 _underlying;
         bool _single_partition;
         bool _static_row_done = false;
+        bool _partition_is_open = false;
```

Comment:
Can't we reuse the existing `_static_row_done` flag (maybe rename it to `_partition_header_done`)? The underlying forwarding reader will always emit the partition header (partition start and static row if any) first, then require fast forwarding to proceed.
I'm fine with 2 flags too if you think it helps code clarity.

Reply:
> Can't we reuse the existing `_static_row_done` flag (maybe rename it to `_partition_header_done`)? The underlying forwarding reader will always emit the partition header (partition start and static row if any) first, then require fast forwarding to proceed.
> I'm fine with 2 flags too if you think it helps code clarity.

Reader should emit a `partition_end` only if there was a `partition_start` before. `_static_row_done` only tells us that the call to `fast_forward_to` was made, is says nothing about the content of the stream, it could be empty.